app/src/main/java/com/example/watpool/services/firebase_services/FirebaseTripsService.kt

```diff
@@ -0,0 +1,203 @@
+package com.example.watpool.services.firebase_services
+
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.example.watpool.services.interfaces.TripsService
+import com.google.android.gms.tasks.Task
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.QuerySnapshot
+import com.google.firebase.firestore.firestore
+import java.time.LocalDate
+import java.util.UUID
+
+class FirebaseTripsService: TripsService {
+    private val database = Firebase.firestore
+    private val tripsRef: CollectionReference = database.collection("trips")
+    private val tripsConfirmationRef: CollectionReference = database.collection("trips_confirmation")
+
+    enum class DayOfTheWeek(val day: String) {
+        SUNDAY ("SUNDAY"), MONDAY ("MONDAY"), TUESDAY ("TUESDAY"),
+        WEDNESDAY ("WEDNESDAY"), THURSDAY ("THURSDAY"), FRIDAY ("FRIDAY"), SATURDAY ("SATURDAY")
+    }
+
+    // fetches the last trip and creates a clone
+//    override fun cloneTrip(tripId: String): Task<DocumentReference> {
+//        val oldTrip = tripsRef.whereEqualTo("id", tripId).get()
+//        return oldTrip
+//    }
+
+    // driver posting a new trip
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun createTrip(driverId: String, startingCoordinateId: String, endingCoordinateId: String, tripDate: LocalDate, maxPassengers: String, isRecurring: Boolean, recurringDayOfTheWeek: DayOfTheWeek, recurringEndDate: LocalDate): Task<DocumentReference> {
+        val id: String = UUID.randomUUID().toString()
+        val trip = hashMapOf(
+            "id" to id,
+            "starting_coordinate" to startingCoordinateId,
+            "ending_coordinate" to endingCoordinateId,
+            "max_passengers" to maxPassengers,
+            "is_recurring" to isRecurring,
+            "trip_date" to tripDate.toString(),
+            "driver_id" to driverId
+        )
+
+        if (isRecurring) {
+            trip["recurring_day"] = recurringDayOfTheWeek.day
+            trip["recurring_end_dat"] = recurringEndDate.toString()
+        }
+
+        return tripsRef.add(trip)
+    }
+
+    // rider requesting new ride
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun createTripPosting(userId: String, startingCoordinateId: String, endingCoordinateId: String, tripDate: LocalDate, isRecurring: Boolean, recurringDayOfTheWeek: DayOfTheWeek, recurringEndDate: LocalDate): Task<DocumentReference> {
+        val id: String = UUID.randomUUID().toString()
+        val trip = hashMapOf(
+            "id" to id,
+            "starting_coordinate" to startingCoordinateId,
```

Comment:
can we get a named location if possible? listing the coordinates for trip details is not a good user experience

Reply:
Same for ending coordinate

Reply:
Can we name the location names for starting and ending coordinates "to" and "from" respectively? ty!